From d6e9623875de5714e85f7da5782ea665116f86dc Mon Sep 17 00:00:00 2001
From: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date: Wed, 30 Mar 2011 01:00:47 +0200
Subject: [PATCH:xf86-input-mouse] Fix OSMouse OS-defined protocol support

Fix regression introduced by 0a088df6: in the case of an OS-specific
protocol, the protocol is PROT_UNKNOWN, but should not be rejected: the core
mouse drive just needs to let the OS driver handle it.

Signed-off-by: Samuel Thibault <samuel.thibault@ens-lyon.org>
---
 src/mouse.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/mouse.c b/src/mouse.c
index edb998f..c3498ea 100644
--- a/src/mouse.c
+++ b/src/mouse.c
@@ -953,6 +953,8 @@ MousePreInit(InputDriverPtr drv, InputInfoPtr pInfo, int flags)
         MouseFindDevice(pInfo, protocol);
 
     xf86Msg(protocolFrom, "%s: Protocol: \"%s\"\n", pInfo->name, protocol);
+    if (protocolID == PROT_UNKNOWN)
+        goto out;
     if (!(pProto = GetProtocol(protocolID)))
     {
 	rc = BadValue;
-- 
1.7.3.2

